pkg/controller/vsphere/reconciler.go

```diff
@@ -389,6 +390,14 @@ func clone(s *machineScope) error {
 	var deviceSpecs []types.BaseVirtualDeviceConfigSpec
 	deviceSpecs = append(deviceSpecs, networkDevices...)
 
+	extraConfig := []types.BaseOptionValue{}
+
+	extraConfig = append(extraConfig, IgnitionConfig(userData)...)
+	extraConfig = append(extraConfig, &types.OptionValue{
+		Key:   GuestInfoHostname,
+		Value: s.machine.GetName(),
```

Comment:
why is this not the same than https://github.com/openshift/machine-api-operator/commit/eb9152c368e62bed358f2aacc3e5d67707361a7d#diff-11b6610f82e5d47fdfb8dda44b90dd1eR283

Reply:
@enxebre the vm doesn't exist yet.  Is that not the name of the machine that will be created?

Reply:
right, It's passed here so should be https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/reconciler.go#L425
It'd be nice to use the same reference consistently everywhere though.